src/Generating/PrefabPiecePool.cpp

```diff
@@ -5,6 +5,81 @@
 
 #include "Globals.h"
 #include "PrefabPiecePool.h"
+#include "../Bindings/LuaState.h"
+#include "SelfTests.h"
+#include "WorldStorage/SchematicFileSerializer.h"
+
+
+
+
+
+// Conditionally log a warning
+#define CONDWARNING(ShouldLog, ...) \
+	if (ShouldLog) \
+	{ \
+		LOGWARNING(__VA_ARGS__); \
+	}
+
+
+
+
+
+#if SELF_TEST
+static class cPrefabPiecePoolTest
+{
+public:
+	cPrefabPiecePoolTest(void)
+	{
+		cSelfTests::Get().Register(cSelfTests::SelfTestFunction(cPrefabPiecePoolTest::TestLoading), "PrefabPiecePool loading test");
+	}
+
+	static void TestLoading(void)
+	{
+		cPrefabPiecePool test;
+		auto res = test.LoadFromFile("test.cubeset", true);
+		if (!res)
+		{
+			LOGWARNING("Loading from file \"test.cubeset\" failed.");
+			return;
+		}
+		LOG("Loaded %u pieces and %u starting pieces", static_cast<unsigned>(test.m_AllPieces.size()), static_cast<unsigned>(test.m_StartingPieces.size()));
+	}
+} g_Test;
```

Comment:
BlockArea could be changed to use dependency injection for the block handlers, and it doesn't look like it has much else in the way of external decencies. The LuaState's plugin manager dependency can be eliminated by by moving Command Result out of PluginManager, and into its own enum class. I'll do that if you want.